ReducedSQLConnectivityChecker.ps1

```diff
@@ -0,0 +1,296 @@
+    }
+}
+
+if (!$(Get-Command 'Resolve-DnsName' -errorAction SilentlyContinue)) {
+    function Resolve-DnsName {
+        param(
+            [Parameter(Position = 0)] $Name,
+            [Parameter()] $Server,
+            [switch] $CacheOnly,
+            [switch] $DnsOnly,
+            [switch] $NoHostsFile
+        );
+        process {
+            # ToDo: Add support
+            Write-Host "WARNING: Current environment doesn't support multiple DNS sources."
+            return @{ IPAddress = [Dns]::GetHostAddresses($Name).IPAddressToString };
+        }
+    }
+}
+
+if (!$(Get-Command 'Get-NetAdapter' -errorAction SilentlyContinue)) {
+    function Get-NetAdapter {
+        param(
+            [Parameter(Position = 0, Mandatory = $true)] $HostName,
+            [Parameter(Mandatory = $true)] $Port
+        );
+        process {
+            Write-Host 'Unsupported'
+        }
+    }
+}
+
+if (!$(Get-Command 'netsh' -errorAction SilentlyContinue) -and $CollectNetworkTrace) {
+    Write-Host "WARNING: Current environment doesn't support network trace capture. This option is now disabled!"
+    $CollectNetworkTrace = $false
+}
+
+# PowerShell Container Image Support End
+
+function IsManagedInstance([String] $Server) {
+    return [bool]((($Server.ToCharArray() | Where-Object { $_ -eq '.' } | Measure-Object).Count) -ge 4)
+}
+
+function IsManagedInstancePublicEndpoint([String] $Server) {
+    return [bool]((IsManagedInstance $Server) -and ($Server -match '.public.'))
+}
+
+function SendAnonymousUsageData {
+    try {
+        #Despite computername and username will be used to calculate a hash string, this will keep you anonymous but allow us to identify multiple runs from the same user
+        $StringBuilderHash = [System.Text.StringBuilder]::new()
+        
+        $text = $env:computername + $env:username
+        if ([string]::IsNullOrEmpty($text)) {
+            $text = $Host.InstanceId
+        }
+        
+        [System.Security.Cryptography.HashAlgorithm]::Create("MD5").ComputeHash([System.Text.Encoding]::UTF8.GetBytes($text)) | ForEach-Object {
+            [Void]$StringBuilderHash.Append($_.ToString("x2"))
+        }
+
+        $body = New-Object PSObject `
+        | Add-Member -PassThru NoteProperty name 'Microsoft.ApplicationInsights.Event' `
+        | Add-Member -PassThru NoteProperty time $([System.dateTime]::UtcNow.ToString('o')) `
+        | Add-Member -PassThru NoteProperty iKey "a75c333b-14cb-4906-aab1-036b31f0ce8a" `
+        | Add-Member -PassThru NoteProperty tags (New-Object PSObject | Add-Member -PassThru NoteProperty 'ai.user.id' $StringBuilderHash.ToString()) `
+        | Add-Member -PassThru NoteProperty data (New-Object PSObject `
+            | Add-Member -PassThru NoteProperty baseType 'EventData' `
+            | Add-Member -PassThru NoteProperty baseData (New-Object PSObject `
+                | Add-Member -PassThru NoteProperty ver 2 `
+                | Add-Member -PassThru NoteProperty name '1.4'));
```

Comment:
Let's make sure we have the same version as "main" here, maybe a differentiator so we can understand how many runs using reduced version.

Reply:
There's a possibility that the 'main' version is also at 1.4 and not 1.5, this is taken from it, but I agree that something like 1.5r should be used here, the idea is to have this developed together with the main script and potentially at some point in time merge it somehow with the main script so we don't have the same code repeated twice for the environments that require a reduced version of the script. This version number should probably be a variable within the script, it's really easy to miss out on it in code.